src/main/java/org/cactoos/io/LoggingInputStream.java

```diff
@@ -0,0 +1,236 @@
+        final int byts = this.origin.read(buf, offset, len);
+        final Instant end = Instant.now();
+        final long millis = Duration.between(start, end).toMillis();
+        final Level level = this.logger.getLevel();
+        final String msg = "Read %d byte(s) from %s in %dms.";
+        if (byts > 0) {
+            this.bytes.getAndAdd(byts);
+            this.time.getAndAdd(millis);
+            if (!level.equals(Level.INFO)) {
+                this.logger.log(
+                    level,
+                    String.format(
+                        msg,
+                        this.bytes.get(),
+                        this.source,
+                        this.time.get()
+                    )
+                );
+            }
+        } else {
+            if (level.equals(Level.INFO)) {
+                this.logger.info(
+                    String.format(
+                        msg,
+                        this.bytes.get(),
+                        this.source,
+                        this.time.get()
+                    )
+                );
+            }
+        }
+        return byts;
+    }
+
+    @Override
+    public long skip(final long num) throws IOException {
+        final long skipped = this.origin.skip(num);
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Skipped %d byte(s) from %s.",
+                skipped,
+                this.source
+            )
+        );
+        return skipped;
+    }
+
+    @Override
+    public int available() throws IOException {
+        final int avail = this.origin.available();
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "There is(are) %d byte(s) avaliable from %s.",
+                avail,
+                this.source
+            )
+        );
+        return avail;
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.origin.close();
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Closed input stream from %s.",
+                this.source
+            )
+        );
+    }
+
+    @Override
+    public void mark(final int limit) {
+        this.origin.mark(limit);
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Marked position %d from %s.",
+                limit,
+                this.source
+            )
+        );
+    }
+
+    @Override
+    public void reset() throws IOException {
+        this.origin.reset();
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Reseted input stream from %s.",
+                this.source
+            )
+        );
+    }
+
+    @Override
+    public boolean markSupported() {
+        final boolean supported = this.origin.markSupported();
+        if (supported) {
```

Comment:
@fabriciofx I think you can get rid of this big ``if()`` by having a smaller one where you build the message, something like this;

```java
final String message;
if(this.origin.markSupported()) {
    message = "Mark and reset are supported";
} else {
    message = "Mark and reset NOT supported";
}
this.logger.log(...)
```

what do you think?

Reply:
@amihaiemil I think it better. I'll change it. Thanks!